articles/react-pwa/index.md

```diff
@@ -0,0 +1,539 @@
+
+- When the state of the form changes, the method `inputTextHandler` bound on the input tag is invoked to update the state to the current input value by calling the `setInputText` method.
+
+- The form is a controlled component that will track the `inputText`. 
+
+- When the submit event is fired on the form, the function `submitTodoHandler` on the `Add` button is executed. We are checking that the input is not empty before creating a new todo in our `addTodo` function passed down from `App` component.
+
+- The method `setInputText` inside the `submitTodoHandler` will reset the form.
+
+The todo will be an object with a structure like:
+```javascript
+{
+  id: v4(),
+  task,
+  complete: false
+}
+```
+
+The ID is a unique identifier or key for a specific todo task. I am using the `uuid` library to generate a unique id.
+
+You can install the `uuid` package using this command in your terminal:
+
+```bash
+npm install uuid
+```
+
+### Creating the TodoList Component
+This is the component that acts as a container to the list of the todo items.
+
+You need to create a new file named `TodosList.js` inside the components folder. Since our Todo component is a child of the TodoList component, we will first need to create the TodoList.
+
+The TodoList component is recieving todos state from the App component, which we will create later. We have [lifted its state up](https://reactjs.org/docs/lifting-state-up.html) in the App component so that our `TodoList` can access it via props.
+
+To create this component, go into your components folder and create a new file named `TodoList.js`.
+
+This component code is like:
+```javascript
+import React from "react";
+import Todo from "./Todo";
+
+const TodoList = ({ todos, setTodos }) => {
+  const completeTodo = (id) => {
+      let updatedTodos=todos.map(todo=>{
+          if(todo.id===id){
+              todo.completed=!todo.completed;
+          }
+          return todo;
+      })
+      setTodos(updatedTodos);
+  };
+  const removeTodo=(id)=>{
+    setTodos(todos.filter(todo=>todo.id!==id));
+  }
+
+  return (
+    <div className="todo-container">
+      <ul className="todo-list">
+        {todos.map((todo) => (
+          <Todo key={todo.id}
+            todo={todo}
+            completeTodo={completeTodo} 
+            removeTodo={removeTodo}/>
+        ))}
+      </ul>
+    </div>
+  );
+};
+
+export default TodoList;
+```
+
+In this component:
+
+- The `TodoList` is receiving `todos` as an array of objects which are accessed by destructuring its props.
+
+- In the return of our function, we map through the `todos` and return a JSX syntax with a new Todo component as the todo item.
+
+- The `removeTodo` passes an `id` that uniquely identifies the `todo` in our list. Whatever todo that does not match the provided `id` is filtered and stored in the state using `setTodos`.
```

Comment:
`The removeTodo passes an id`

You should say that the `removeTodo` is a function and it accepts an ID.